charts/kyverno/templates/networkpolicy.yaml

```diff
@@ -30,4 +30,78 @@ spec:
   ingress:
     - {}
   {{- end }}
+---
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: allow-prometheus
+  namespace: {{ include "kyverno.namespace" . }}
+  labels:
+    app: kyverno
+spec:
+  podSelector:
+    matchLabels:
+      app: kyverno
+  ingress:
+  - from:
+    {{- with .Values.networkPolicy.prometheusNamespaceLabels }}
+    - namespaceSelector:
+        matchLabels:
+            {{ toYaml . | nindent 10 }}
+    {{- end }}
+    {{- with .Values.networkPolicy.prometheusPodLabels }}
+      podSelector:
+        matchLabels:
+            {{ toYaml . | nindent 10 }}
+    {{- end }}
+    ports:
+    # Webhook access
+    - protocol: TCP
+      port: 9443
+    # Allow prometheus scrape for metrics
+    - protocol: TCP
+      port: 8000
+  policyTypes:
+    - Ingress
+    - Egress
+---
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: allow-apiserver
+  namespace: {{ include "kyverno.namespace" . }}
+spec:
+  podSelector: {}
+  policyTypes:
+  - Ingress
+  - Egress
+  # Allow ingress from kube-apiserver on port 9443
+  ingress:
+    - from:
+        - namespaceSelector:
+            matchLabels:
+              kubernetes.io/metadata.name: kube-system
+          podSelector:
+            matchLabels:
+              component: kube-apiserver
+              tier: control-plane
+      ports:
+        - port: 9443
+  # If openEgress is true allow all egress traffic else allow egress to only kube-apiserver on port 9443
+  {{- if .Values.networkPolicy.openEgress }}
+  egress:
+  - {}
+  {{- else }}
+  egress:
+    - to:
+        - namespaceSelector:
+            matchLabels:
+              kubernetes.io/metadata.name: kube-system
+          podSelector:
+            matchLabels:
+              component: kube-apiserver
+              tier: control-plane
+      ports:
+        - port: 9443
```

Comment:
I think the api server port is 6443

Reply:
Just FYI this port is referring to webhook port on Kyverno

Reply:
But the rule is an egress rule. Egress is traffic from the kyverno pod to the api server and the port is the target port (in this case the port of the kubernetes api server).

Reply:
Yep you are correct, so this should be Kubernetes API server port

Reply:
Ok, so I will change this to 6443..

Reply:
I would imagine managed Kubernetes like EKS or GKS might use different ports.

Reply:
So then, we have to mention this in the docs .?

Reply:
Most likely.  For Helm I think making the port number something that can be changed is good too.

Reply:
Sounds good, will change those too then...

Reply:
In managed clusters the apiserver pods are not visible and you can’t really identify the traffic using labels isn’t it ?